app/main.py

```diff
@@ -1,2 +1,242 @@
+from __future__ import annotations
+
+import copy
+import dataclasses
+from typing import Any, Callable, Iterable, Hashable
+
+
+@dataclasses.dataclass
+class Node:
+    key: Hashable
+    k_hash: int
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        # added mainly to keep track of input order, iteration
+        self._keys = []
+        self._values = []
+
+        self._initial_capacity = 8
+        self._load_factor = 2 / 3
+
+        self._capacity = self._initial_capacity
+        self._hash_table = [[] for _ in range(self._capacity)]
```

Comment:
Why not simplify this by filling hash_table with None values? It will be more efficient then creating one thousand empty lists. Also can you annotate hash table so it actually shows that it is a list of Nodes